### ISSUE TITLE
[ue] Unable to import skeleton data with the `Inherit` timeline keys in the middle of animation.

### ISSUE THREAD

**Issue by misaki-eymard** [EsotericSoftware/spine-runtimes#2672]
**Description:**
With the `Inherit` timeline available since Spine 4.2, switching inheritance on or off in the middle of the animation will prevent the skeleton data from being imported into UE. (Attempting to import will result in infinite loading.)

(The problem does not occur if there is the `Inherit` timeline but the inheritance is toggled on and off at frame 0 of the animation.)

**Expected behavior:**
Even skeletons that switch inheritance in the middle of an animation can be imported successfully.

**Steps to reproduce:**
1. Download the attached .zip file: [Unable to import skeleton data with the `Inherit` timeline keys in the middle of animation.zip](https://github.com/user-attachments/files/17654987/Unable.to.import.skeleton.data.with.the.Inherit.timeline.keys.in.the.middle.of.animation.zip)
1. Open a UE project with the spine-ue runtime installed and import the file `spineboy-pro_should-cause-crash.json` under `Skeleton-should-cause-crash`:
![Screenshot 2024-11-06 at 15 52 45](https://github.com/user-attachments/assets/9e719888-5933-4dad-92da-d8c3c676b478)
1. You should see the import not finish forever or the UE editor will crash.

For reference, a skeleton with some inherit keys only at frame 0 of the animation is included in the same zip as `Skeleton-should-be-imported-correctly`.

**The version that can reproduce this issue:**
UE5.4.4

**This issue was reported in the following thread on the forum:**
https://esotericsoftware.com/forum/d/27218-skeletons-with-inherit-crash-the-editor

**Comment by badlogic** [comment]
<img width="1120" alt="Screenshot 2024-11-06 at 13 33 34" src="https://github.com/user-attachments/assets/57fc6875-9925-4117-a310-b557aacb2977">

Can not reproduce using latest spine-cpp and spine-ue from the 4.2 branch and UE 5.4.4. I suppose the user hasn't updated their copy of spine-cpp.

**Comment by badlogic** [comment]
Creating an actor with Spine components and playing back either `inherit-enable` or `inherit-disable` also works as expected.

<img width="3008" alt="Screenshot 2024-11-06 at 13 41 29" src="https://github.com/user-attachments/assets/b94ecd09-83d1-459c-a51c-9dbe660a0873">

**Comment by misaki-eymard** [comment]
Sorry, the reproduction project I attached was bad. I oversimplified the animation of the `Inherit` timeline. This JSON file should be able to reproduce the crash: 
[spineboy-pro_should-cause-crash.json](https://github.com/user-attachments/files/17654983/spineboy-pro_should-cause-crash.json)

I added a new key at frame 0. The `Inherit` key at frame 0 enables rotation inheritance (this is the same as the setup pose, so the animation looks the same as the previous one), and the key at frame 30 disables rotation inheritance. Attempting to import a skeleton with this animation causes infinite loading.

I have also replaced the .zip file attached above with the new one.
